model/src/main/java/org/projectnessie/api/ConfigApi.java

```diff
@@ -17,6 +17,7 @@
 
 import org.projectnessie.model.NessieConfiguration;
 
+@Deprecated
```

Comment:
Im not sure I understand. This statement deprecates an entire class hierarchy because one of the leaf nodes will be removed?

Im also not sure why the Tree/Contents etc would go away? They still exist in the server side and any auto-generated client eg typescript would use them too

Reply:
Only `TreeApi` + `ContentsApi` + `ConfigApi` as "user visible" APIs would go away once the v0.9 API is removed.

`HttpConfigApi`, `HttpContentsApi`, `HttpTreeApi` will stay as the contract for `*Resource` classes, generated UI classes, openapi spec, etc. `HttpConfigApi`, `HttpContentsApi`, `HttpTreeApi` then get e.g. the `extends ContentsApi` removed.

So users of the 1.0 API don't see all the REST/HTTP details.

Reply:
perfect, that was my expectation. Which is why the `Deprecated` looks so weird. We aren't deprecating/removing anything we are just changing the client public API.

Reply:
Right. v0.9 will become "deprecated for removal" and v1.0 _the_ API (until we might have some v2.0 for some reason).